cmd/influxd/restore/restore.go

```diff
@@ -227,41 +367,108 @@ func (cmd *Command) unpackShard(shardID string) error {
 	}
 
 	// find the shard backup files
-	pat := filepath.Join(cmd.backupFilesPath, fmt.Sprintf(backup_util.BackupFilePattern, cmd.database, cmd.retention, id))
+	pat := filepath.Join(cmd.backupFilesPath, fmt.Sprintf(backup_util.BackupFilePattern, cmd.destinationDatabase, cmd.restoreRetention, id))
 	return cmd.unpackFiles(pat + ".*")
 }
 
-// unpackDatabase will look for all backup files in the path matching this database
+// unpackFiles will look for backup files matching the pattern and restore them to the data dir
+func (cmd *Command) uploadShardsLive() error {
+	// find the destinationDatabase backup files
+	if cmd.enterprise {
+		for _, file := range cmd.manifest.Files {
+			if cmd.sourceDatabase == "" || cmd.sourceDatabase == file.Database {
+				if cmd.backupRetention == "" || cmd.backupRetention == file.Policy {
+					if cmd.shard == 0 || cmd.shard == file.ShardID {
+						cmd.StdoutLogger.Printf("Restoring shard %d live from backup %s\n", file.ShardID, file.FileName)
+						f, err := os.Open(filepath.Join(cmd.backupFilesPath, file.FileName))
+						if err != nil {
+							return err
+						}
+						gr, err := gzip.NewReader(f)
+						if err != nil {
+							return err
+						}
+						tr := tar.NewReader(gr)
+						targetDB := cmd.destinationDatabase
+						if targetDB == "" {
+							targetDB = file.Database
+						}
+
+						if err := cmd.client.UploadShard(file.ShardID, cmd.shardIDMap[file.ShardID], cmd.destinationDatabase, cmd.restoreRetention, tr); err != nil {
+							return err
+						}
+						f.Close()
+					}
+				}
+			}
+		}
+	} else {
```

Comment:
Split this off somehow. Two giant blocks of code split by a single `else` makes it difficult to maintain.